dashboard/app/controllers/projects_controller.rb

```diff
@@ -158,14 +158,26 @@ def create_new
     redirect_to action: 'edit', channel_id: ChannelToken.create_channel(
       request.ip,
       StorageApps.new(storage_id('user')),
-      data: {
-        name: 'Untitled Project',
-        level: polymorphic_url([params[:key], 'project_projects'])
-      },
+      data: initial_data,
       type: params[:key]
     )
   end
 
+  private def initial_data
+    data = {
+      name: 'Untitled Project',
+      level: polymorphic_url([params[:key], 'project_projects'])
+    }
+
+    # We do not currently generate thumbnails for flappy, so specify a
+    # placeholder image here. This allows flappy projects to show up in the
+    # public gallery, and to be published from the share dialog.
+    if params[:key] == 'flappy'
+      data[:thumbnailUrl] = '/blockly/media/flappy/placeholder.jpg'
+    end
```

Comment:
It feels a little unfortunate having app-specific code live here. Could we perhaps leverage STANDALONE_PROJECTS and do something like
```
data[:thumbnailUrl] = STANDALONE_PROJECTS[params[:key]][:fallbackThumbnailUrl]
```
and just have fallbackThumbnailUrl be undefined for everything other than flappy?

Not sure if that's an abuse of how STANDALONE_PROJECTS is currently being used.

Reply:
If that doesn't make sense, the current approach isn't that bad.

Reply:
I like where you're going with this. I think that could work if we improve the naming and documentation in/on STANDALONE_PROJECTS.